great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1302,88 +1287,40 @@ def compute_metrics(self, metrics: Metric | list[Metric]) -> MetricResult | list
             this by adding an explicit annotation (e.g. metrics: list[Metric] = ...)
             https://github.com/python/mypy/issues/18712
         """
+        is_single_metric = False
```

Comment:
I keep track of this so I can tell whether `[metric]` or just `metric` was passed in so we can return `[metric_result]` or `metric_result`. Previously we just checked for the length of the input which doesn't distinguish these.